### ISSUE TITLE
Modular encoding, `-e 9` slower than `-e 10` with `--num_threads 0`

### ISSUE THREAD

**Issue by TheHardew** [libjxl/libjxl#3323]
**Describe the bug**
`-e 9` is slower than `-e 10` with `--num_threads 0` with lossless encoding

**To Reproduce**
```
hyperfine 'cjxl -e 9 -d 0 --num_threads 0 img.png /dev/null' 'cjxl -e 10 -d 0 --num_threads 0 img.png /dev/null'
Benchmark 1: cjxl -e 9 -d 0 --num_threads 0 img.png /dev/null
  Time (mean ± σ):     36.188 s ±  0.217 s    [User: 36.117 s, System: 0.033 s]
  Range (min … max):   35.786 s … 36.494 s    10 runs
 
Benchmark 2: cjxl -e 10 -d 0 --num_threads 0 img.png /dev/null
  Time (mean ± σ):     32.581 s ±  0.243 s    [User: 32.398 s, System: 0.143 s]
  Range (min … max):   32.240 s … 32.983 s    10 runs
 
Summary
  cjxl -e 10 -d 0 --num_threads 0 img.png /dev/null ran
    1.11 ± 0.01 times faster than cjxl -e 9 -d 0 --num_threads 0 img.png /dev/null
```
**Expected behavior**
`-e 9` should be faster than `-e 10`

**Screenshots**
Input file:
![img](https://github.com/libjxl/libjxl/assets/7374061/6f11d613-485a-4b39-a35c-0e644b4163a0)

**Environment**
 - OS: Arch Linux
 - Compiler version: clang 16.0.6
 - CPU type: x86_64 Ryzen 2700x
 - cjxl/djxl version string: cjxl v0.10.0 b38469de [AVX2]

**Comment by veluca93** [comment]
I believe https://github.com/libjxl/libjxl/pull/3337 will also fix this issue... Would you be able to confirm?

**Comment by TheHardew** [comment]
Sure, I'll build it and test

**Comment by TheHardew** [comment]
| preset | size [kB] | time |
|-|-|-|
|main `-e 9` | 1073.8 | 36.2|
| main `-e 10` | 956.8 | 32.5|
| main `-e9 --patches 1` | 956.7|31.4|
| #3337 `-e 9` | 1192.1 | 7.7 |
| #3337 `-e 9 --patches1` | 1055.0 | 8.6 |

**Comment by veluca93** [comment]
> preset	size [kB]	time
> [0d2e3ed](https://github.com/libjxl/libjxl/commit/0d2e3edccfb41bfdd3c8d10423b8eaadeecf9477) `-e 9`	1073.8	36.2
> [0d2e3ed](https://github.com/libjxl/libjxl/commit/0d2e3edccfb41bfdd3c8d10423b8eaadeecf9477) `-e 10`	956.8	32.5
> [0d2e3ed](https://github.com/libjxl/libjxl/commit/0d2e3edccfb41bfdd3c8d10423b8eaadeecf9477) `-e9 --patches 1`	956.7	31.4
> #3337 `-e 9`	1192.1	7.7
> #3337 `-e 9 --patches1`	1055.0	8.6

That sounds good - is it still smaller than `-e 8`?

**Comment by TheHardew** [comment]
| preset | size [kB] | time |
|-|-|-|
|#3337 `-e8` | 1200.3 | 3.6 |
|#3337 `-e8 --patches 1` | 1085.8 | 4.6 |

though I am testing just one image  
all times with `--num_threads 0`

**Comment by veluca93** [comment]
Good, I'll consider it be fixed by #3337 then.
